packages/svelte/src/compiler/phases/2-analyze/index.js

```diff
@@ -193,13 +193,23 @@ function get_delegated_event(event_name, handler, context) {
 			binding !== null &&
 			// Bail-out if the the binding is a rest param
 			(binding.declaration_kind === 'rest_param' ||
-				// Bail-out if we reference anything from the EachBlock (for now) that mutates in non-runes mode,
-				(((!context.state.analysis.runes && binding.kind === 'each') ||
-					// or any normal not reactive bindings that are mutated.
-					binding.kind === 'normal' ||
-					// or any reactive imports (those are rewritten) (can only happen in legacy mode)
-					binding.kind === 'legacy_reactive_import') &&
-					binding.mutated))
+				// Bail-out if we reference anything from the EachBlock (for now) that mutates.
+				(binding.kind === 'each' &&
```

Comment:
I vaguely remember that Rich wanted this kind of mutation to the each entry directly to be disallowed in runes mode - but I'm not sure if this was more due to the fact that this wasn't possible to achieve correctly before we added the proxy, so may be ok now. @Rich-Harris do you remember?

Reply:
Ah yeah. Well we can go that route if needed instead.

Reply:
Changed the PR to reflect that instead.

Reply:
We can always revert that commits lol

Reply:
I think it should be disallowed, because otherwise things like this break down:

```diff
-{#each numbers as n}
+{#each numbers.filter((n) => n % 2 === 0) as n}
  <button onclick={() => n += 1}>{n}</button>
{/each}
```

It's better to enforce consistency (e.g. 'you have to do `numbers[i] += 1`') than to have something that is superficially more convenient, but which is pretty confusing on the face of it (it's basically like reassigning a function parameter, and yet it somehow works?) and which requires you to understand non-obvious properties of the system before you can wield it successfully.

[It doesn't work in Svelte 4, FWIW](https://svelte.dev/repl/62e07b0e756245a3a7ba3f65405fa83f?version=4.2.10), so we should disallow it regardless of runes mode. I think it would be better to do this work in `AssignmentExpression` and `UpdateExpression` validation visitors — lemme push up some code real quick

Reply:
lol ok ignore the svelte 4 comment, i was doing `onclick` instead of `on:click` 🤦

Reply:
though that said, [this](https://svelte.dev/repl/62e07b0e756245a3a7ba3f65405fa83f?version=4.2.10) is a useful demo of 'works until it doesn't'

Reply:
Hey @Rich-Harris what exactly doesn’t work in this demo/examples?